### ISSUE TITLE
Idea(s) for multi-scheme models

### ISSUE THREAD

**Issue by ndgnuh** [ndgnuh/megane#3]
- Input processing is just 1 function: resize image
- Each module in the modeling handles their label encoding separately
- Trainer doesn't care about label encoding and middle metrics
- Sample is input by the trainer and output by the model, metric is calculated end2end (based on the `Sample`)

**Comment by ndgnuh** [comment]
- [ ] should dataloader collate returns raw data?

**Comment by ndgnuh** [comment]
```python
        for step, (images, samples) in enumerate(loader):
            optimizer.zero_grad()
            outputs: FixedOutputType = model(images, post_process=True)
            for i, sample in enumerate(samples)
                sample.boxes = outputs.boxes[i]
                sample.classes = outputs.classes[i]
                sample.scores = outputs.scores[i]
            fabric.backward(outputs.loss)
            fabric.clip_gradients(model, optimizer, max_norm=5)
            optimizer.step()
```

where `FixedOutputType` is a convention by another modules (not trainer or model)
